sigstore/_cli.py

```diff
@@ -312,11 +320,19 @@ def _verify(
     rekor_url: str,
     staging: bool,
 ) -> None:
-    # If the user has explicitly requested the staging instance,
-    # we need to override some of the CLI's defaults.
     if staging:
         logger.debug("verify: staging instances requested")
-        rekor_url = STAGING_REKOR_URL
+        verifier = Verifier.staging()
+    elif rekor_url == DEFAULT_REKOR_URL:
+        verifier = Verifier.production()
+    else:
+        # TODO: We need CLI flags that allow the user to figure the Fulcio cert chain
+        # for verification.
+        click.echo(
+            "Custom Rekor and Fulcio configuration for verification isn't fully supported yet!",
+            err=True,
+        )
+        raise click.Abort
```

Comment:
Flagging for review: we don't really have a clean way to do this yet, since it would require yet another flag (one that can be repeated) to add custom certificates for the custom Fulcio instance's chain.

I believe `cosign` sidesteps this entirely by relying heavily on TUF, so this will sort of resolve itself when #25 is resolved. Even then, however, the user will need a way to specify custom TUF root metadata so that we can bootstrap the TUF client, and the TUF repository will need to contain the various files at consistent locations/targets.